libevmasm/AssemblyItem.cpp

```diff
@@ -40,13 +42,15 @@ AssemblyItem AssemblyItem::toSubAssemblyTag(size_t _subId) const
 pair<size_t, size_t> AssemblyItem::splitForeignPushTag() const
 {
 	assertThrow(m_type == PushTag || m_type == Tag, Exception, "");
-	return make_pair(size_t((data()) / (u256(1) << 64)) - 1, size_t(data()));
+	// There may be a bug in boost where typecasting u256 to size_t results in all bits set if the higher bit was set
+	return make_pair(size_t((data() >> 64) - 1), size_t(data() & 0xffffffffffffffff));
```

Comment:
The shift definitely feels more readable, the masking seems superfluous, but it seems on some boost / compiler combinations this was misbehaving and returning all bits set for @aarlt and @bit-shift.

Reply:
Perhaps better to use `u256(data())` to guard against any changes we might make to the return vale of `data()`.

Reply:
Is `0xffffffffffffffff` the right think to use? Don't you need something like `0xffffffffffffffffL`?